docs/BR.md

```diff
@@ -1739,142 +1737,875 @@ The CA SHALL enforce multi-factor authentication for all accounts capable of dir
+| \ \ \ \ `GeneralSubtree`  | The requirements for a `GeneralSubtree` that appears within a `permittedSubtrees`. |
+| \ \ \ \ \ \ \ \ `base`    | See following table. |
+| \ \ \ \ \ \ \ \ `minimum` | MUST NOT be present. |
+| \ \ \ \ \ \ \ \ `maximum` | MUST NOT be present. |
+
+The following table contains the requirements for the `GeneralName` that appears within the `base` of a `GeneralSubtree` in either the `permittedSubtrees` or `excludedSubtrees`.
+
+Table: `GeneralName` requirements for the `base` field
+
+| __Name Type__   | __Presence__ |  __Permitted Subtrees__ | __Excluded Subtrees__ | __Entire Namespace Exclusion__ |
+| ---             | -            | ----                    | ----                  | ----                           |
+| `dNSName`       | MUST         | The CA MUST confirm that the Applicant has registered the `dNSName` or has been authorized by the domain registrant to act on the registrant's behalf. See [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control). | If at least one `dNSName` instance is present in the `permittedSubtrees`, the CA MAY indicate one or more subordinate domains to be excluded. | If no `dNSName` instance is present in the `permittedSubtrees`, then the CA MUST include a zero-length `dNSName` to indicate no domain names are permitted. |
+| `iPAddress`     | MUST        | The CA MUST confirm that the Applicant has been assigned the `iPAddress` range or has been authorized by the assigner to act on the asignee's behalf. See [Section 3.2.2.5](#3225-authentication-for-an-ip-address). | If at least one `iPAddress` instance is present in the `permittedSubtrees`, the CA MAY indicate one or more subdivisions of those ranges to be excluded. | If no IPv4 `iPAddress` is present in the `permittedSubtrees`, the CA MUST include an `iPAddress` of 8 zero octets, indicating the IPv4 range of 0.0.0.0/0 being excluded. If no IPv6 `iPAddress` is present in the `permittedSubtrees`, the CA MUST include an `iPAddress` of 32 zero octets, indicating the IPv6 range of ::0/0 being excluded. |
+| `directoryName` | MUST        | The CA MUST confirm the Applicant's and/or Subsidiary's name attributes such that all certificates issued will comply with the relevant Certificate Profile (see [Section 7.1.2](#712-certificate-content-and-extensions)), including Name Forms (See [Section 7.1.4](#714-name-forms)). | The CA SHOULD NOT include values within `excludedSubtrees`. | The CA MUST include a value within `permittedSubtrees`, and as such, this does not apply. See the Excluded Subtrees requirements for more. |
+| Any other value | SHOULD NOT  | -                        | -                     | -                              |
```

Comment:
MozPol section 1.1 (2), second bullet point has the following:

> name constraints which do not allow Subject Alternative Names (SANs) of any of the following types: dNSName, iPAddress, SRVName, rfc822Name 

I think "SHOULD NOT" is too strong here in light of Mozilla policy, at least until it changes to no longer specify SRVName (https://github.com/mozilla/pkipolicy/issues/96). I recommend using a "MAY" here.

Reply:
I think MAY ends up more permissive than practice. The aforementioned Mozilla policy language is relevant for the scope of their policy (which is a super-set of the BRs). However, as the BRs already prohibit the issuance of both `SRVNames` and `rfc822names` within a BR-compliant hierarchy, any CA should be able to meet the Mozilla policy requirements by using an EKU constraint.

If you have the `id-kp-serverAuth` EKU, and you're technically constrained in line with the BRs, then Mozilla policy still declares you "technically capable", which subjects you to Mozilla's auditing requirements. However, because the BRs define "technically capable" as a subset of that,  you fulfill Mozilla's auditing requirements by virtue of Section 8.

That said, I totally appreciate your point here. I'll add explicit language for these name types (incorporating from Mozilla's requirements), but I think SHOULD NOT is still right because you really shouldn't need to include these.

CC'ing @BenWilson-Mozilla to allow him a chance to review this before tomorrow's call.

Reply:
@CBonnell Marking this as resolved with the updated `otherName` handling, but please feel free to raise issues if you still see them.